### PR TITLE
Update some dependencies

### LINKED ISSUE #3189: [infra] Upgrade Docsy, postcss-cli, and autoprefixer
More recent releases of Docsy do not require recursive submodules, have updated dependencies, and other improvements. Consider upgrading.

### PR DESCRIPTION
**Pull request by Okabe-Junya**
### Describe your changes

- Update some dependencies

### Related issue number or link (ex: `resolves #issue-number`)

ref. #3189 

### Checklist before opening this PR (put `x` in the checkboxes)
- [x] This PR does not contain plagiarism
  - don’t copy other people’s work unless you are quoting and contributing it to them.
- [x] I have signed off on all commits 
  - [signing off](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) (ex: `git commit -s`) is to affirm that commits comply [DCO](https://wiki.linuxfoundation.org/dco). If you are working locally, you could add an alias to your `gitconfig` by running `git config --global alias.ci "commit -s"`.

### CODE REVIEW

File: layouts/partials/head.html [hunk 1]
```diff
@@ -52,7 +52,7 @@
   {{ if hasPrefix .Site.GoogleAnalytics "G-"}}
     {{ template "_internal/google_analytics.html" . }}
   {{ else }}
-    {{ template "_internal/google_analytics_async.html" . }}
+    {{ template "_internal/google_analytics.html" . }}
   {{ end }}
```
Comment (jihoon-seo) [inline layouts/partials/head.html:56]: @Okabe-Junya Thank you!

In your suggestion, L53 (`{{ template "_internal/google_analytics.html" . }}`) takes effect regardless of whether `if` in L52 is true or not.

How about this?

[Option 1: comment out redundant codes]
```
  {{/* if hasPrefix .Site.GoogleAnalytics "G-" */}}
    {{ template "_internal/google_analytics.html" . }}
  {{/* else }}
    {{ template "_internal/google_analytics_async.html" . }}
  {{ end */}}
```

[Option 2: remove redundant codes]
```
  {{ template "_internal/google_analytics.html" . }}
```

(I have not tested those code snippets. We need to check whether those codes actually works without error.)

File: package.json [hunk 2]
```diff
@@ -26,15 +26,16 @@
     "update:pkgs": "npx npm-check-updates -u"
   },
   "dependencies": {
-    "lunr-languages": "^1.9.0"
+    "lunr-languages": "^1.14.0"
```
Comment (cjyabraham) [inline package.json:29]: I think this line can be removed as we don't use lunr for search.

File: layouts/partials/head.html (hunk 1 above)
Comment (Okabe-Junya) [reply layouts/partials/head.html:56]: Thanks!

In GA4, the Async file has been deprecated, and this is now enough to configure. (https://github.com/gohugoio/hugo/issues/10899 is a useful reference)

FYI: https://gohugo.io/templates/embedded/#google-analytics

File: package.json (hunk 2 above)
Comment (Okabe-Junya) [reply package.json]: Thanks, fixed by https://github.com/cncf/glossary/pull/3210/commits/557049345211cdf661c56b54360810b0530d08bc

File: layouts/partials/head.html (hunk 1 above)
Comment (Okabe-Junya) [reply layouts/partials/head.html:56]: Oh, I misunderstood the Option 2 you proposed. Option 2 is the most reasonable; I will fix

Comment (Okabe-Junya) [reply layouts/partials/head.html]: fixed by https://github.com/cncf/glossary/pull/3210/commits/9a968122d34585740284824813b1ce6ddd9c6cdb

**Review by cjyabraham — CHANGES_REQUESTED** [review]
For some reason the [Browse by Tags](https://deploy-preview-3210--cncfglossary.netlify.app/tags/?all=true) page doesn't seem to work properly. When I do "Select All" I only see one results. Could you figure out which change you made caused this?

**Comment by Okabe-Junya** [comment]
Ah, great point! Thank you for bringing that up, @cjyabraham . I've been looking into this bug, and while I don’t have a complete explanation yet, it seems to occur when updating Hugo from v0.122 to [v0.123](https://github.com/gohugoio/hugo/releases/tag/v0.123.0). Since v0.123 includes some breaking changes, it might take a bit more time to fully investigate.

For this PR, I've addressed the bug by stopping the update at v0.122, and it seems to be working well now.

**Review by cjyabraham — APPROVED** [review]

File: layouts/partials/head.html [hunk 3]
```diff
@@ -49,11 +49,7 @@
...
 {{ if hugo.IsProduction }}
-  {{ if hasPrefix .Site.GoogleAnalytics "G-"}}
-    {{ template "_internal/google_analytics.html" . }}
-  {{ else }}
-    {{ template "_internal/google_analytics_async.html" . }}
-  {{ end }}
+  {{ template "_internal/google_analytics.html" . }}
 
   <!-- Google Tag Manager -->
   <script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
```
Comment (chalin) [inline layouts/partials/head.html:55]: @cjyabraham - is GTM still being used by this site?

File: layouts/partials/head.html
Comment (chalin) [inline layouts/partials/head.html:1]: Whenever Docsy is update in a followup PR, consideration should be given to eliminating this partial override, and moving any customization to, say, `layouts/partials/hooks/head-end.html` if that makes sense (or maybe a custom `layouts/partials/script.html` file).

/cc @cjyabraham

**Review by chalin — COMMENTED** [review]
> ### Related issue number or link (ex: `resolves #issue-number`)
> close #3189

This **contributes** to #3189 **_without_ closing** it -- please update the opening comment to reflect this. Issue #3189 is also about _updating Docsy_, which is where the heavy lifting will occur. The current version of Docsy used in this repo is 3 years old.

If @cjyabraham is happy with this update, I'm willing to give it a thumbs up too.

**Comment by Okabe-Junya** [comment]
Thank you for your comment, @chalin! As you suggested, I've removed the link to the Issue, so 3189 won't be closed when this PR is merged.

Additionally, I'd like to merge this PR as it is, separate from the docsy update. I think it’s useful as a maintenance PR, so I'd like to go ahead with merging it, thanks!

File: layouts/partials/head.html (hunk 3 above)
Comment (cjyabraham) [reply layouts/partials/head.html:55]: yes

**Comment by Okabe-Junya** [comment]
/cc @seokho-son @jihoon-seo 

I think this PR is ready for review / merge! Please take another look to move forward, thanks!

**Review by jihoon-seo — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**